src/plugin.ts

```diff
@@ -82,5 +86,9 @@ export const cloudStorage =
 
         return existingCollection
       }),
+      onInit: payload => {
```

Comment:
This is a great PR! The only thing that I can see that needs to be adjusted is here. The `onInit` function can potentially be `async`, so the incoming `config.onInit` should be awaited.

GREAT PR. Seriously. This will be amazing to have.

Reply:
Oh, nice catch

Thank you! :clinking_glasses: